pkg/healthautoexport/types.go

```diff
@@ -61,6 +78,66 @@ type WorkoutFields map[string]*QtyWithUnit
 // https://stackoverflow.com/a/43178272/2037090
 type workoutCopy Workout
 
+func (m *Metric) UnmarshalJSON(bytes []byte) error {
+	type MetricInternal struct {
+		Name  string          `json:"name"`
+		Units Units           `json:"units"`
+		Data  json.RawMessage `json:"data"`
+	}
+	mi := MetricInternal{}
+	if err := jsoniter.Unmarshal(bytes, &mi); err != nil {
+		return err
+	}
+	m.Name = mi.Name
+	m.Units = mi.Units
+	if mi.Name == "sleep_analysis" {
```

Comment:
If we adopt the suggestion above, I think this could be a `switch` block to handle `m.Name`, with the `default` case writing into `Datapoints`